test/e2e/sns/cross-region.sh

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+ROOT_DIR="$THIS_DIR/../../.."
+SCRIPTS_DIR="$ROOT_DIR/scripts"
+
+source "$SCRIPTS_DIR/lib/common.sh"
+source "$SCRIPTS_DIR/lib/k8s.sh"
+source "$SCRIPTS_DIR/lib/aws/sns.sh"
+source "$SCRIPTS_DIR/lib/testutil.sh"
+
+TESTING_NAMESPACE=${TESTING_NAMESPACE:-"testing-$RANDOM"}
+AWS_REGION_OVERRIDE=${AWS_REGION_OVERRIDE:-"eu-west-3"}
+AWS_REGION_DEFAULT=${AWS_REGION_DEFAULT:-"eu-central-1"}
+
+test_name="$( filenoext "${BASH_SOURCE[0]}" )"
+service_name="sns"
+ack_ctrl_pod_id=$( controller_pod_id )
+debug_msg "executing test: $service_name/$test_name"
+
+topic_name="ack-test-smoke-$service_name"
+resource_name="topics/$topic_name"
+topic_arn_override_region="arn:aws:sns:$AWS_REGION_OVERRIDE:$AWS_ACCOUNT_ID:$topic_name"
+topic_arn_default_region="arn:aws:sns:$AWS_REGION_DEFAULT:$AWS_ACCOUNT_ID:$topic_name"
+
+
+# PRE-CHECKS
+if sns_topic_exists "$topic_arn_override_region" "$AWS_REGION_OVERRIDE"; then
+    echo "FAIL: expected $topic_name to not exist in SNS. Did previous test run cleanup?"
```

Comment:
Please include the region in this error message :+1: 

I was super confused about this before I realized what was going on :)

```
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ make kind-test SERVICE=sns AWS_ROLE_ARN=$ROLE_ARN
checking AWS credentials ... ok.
creating kind cluster ack-test-e59251b8-eb1e6e2a ... ok.
building aws-controllers-k8s:sns-2b66606 docker image ... ok.
loading the images into the cluster ... ok.
loading CRD manifests for sns into the cluster ... ok.
loading RBAC manifests for sns into the cluster ... ok.
loading service controller Deployment for sns into the cluster ...ok.
generating AWS temporary credentials and adding to env vars map ... ok.
======================================================================================================
To poke around your test cluster manually:
export KUBECONFIG=/home/jaypipes/go/src/github.com/aws/aws-controllers-k8s/scripts/lib/../../build/tmp-ack-test-e59251b8-eb1e6e2a/kubeconfig
kubectl get pods -A
======================================================================================================
testing Helm release for sns for release version 2b66606.
Building release artifacts for sns-2b66606
Generating custom resource definitions for sns
Generating RBAC manifests for sns
namespace/ack-system-test-helm created
installing the helm chart for ack-sns-controller in namespace ack-system-test-helm ... ok.
uninstalling the helm chart for ack-sns-controller in namespace ack-system-test-helm ... ok.
FAIL: expected ack-test-smoke-sns to not exist in SNS. Did previous test run cleanup?
To resume test with the same cluster use: "-c /home/jaypipes/go/src/github.com/aws/aws-controllers-k8s/scripts/lib/../../build/tmp-ack-test-e59251b8-eb1e6e2a"
make: *** [Makefile:51: kind-test] Error 1
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ daws sns list-topics
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ daws sns list-topics --region=eu-west-3
jaypipes@thelio:~/go/src/github.com/aws/aws-controllers-k8s$ daws sns list-topics --region=eu-central-1
TOPICS	arn:aws:sns:eu-central-1:750630568209:ack-test-smoke-sns
```

Reply:
@A-Hilaly maybe I need to apply https://github.com/aws/aws-controllers-k8s/pull/499 before running this?

Reply:
> Please include the region in this error message 👍

Done. Sorry for the confusion. My bad

Reply:
> @A-Hilaly maybe I need to apply #499 before running this?

@jaypipes After cleaning up the test objects (resource in k8s and AWS), and #499 everything should be OK.

Reply:
> > Please include the region in this error message +1
> 
> Done. Sorry for the confusion. My bad

@A-Hilaly I don't see that you updated the message?

Reply:
@jaypipes Apologies, i might have forgotten to push my changes. Messages fixed and PR rebased.

```
./test/e2e/sns/cross-region.sh
namespace/testing-3897 created
topic.sns.services.k8s.aws/ack-test-smoke-sns created
topic.sns.services.k8s.aws "ack-test-smoke-sns" deleted
topic.sns.services.k8s.aws/ack-test-smoke-sns created
topic.sns.services.k8s.aws "ack-test-smoke-sns" deleted
namespace "testing-3897" deleted

```